draft-ietf-tls-esni.md

```diff
@@ -726,13 +742,24 @@ MAY offer to resume sessions established without ECH.
 
 # Server Behavior {#server-behavior}
 
+Servers that implement the ECH extension play one of two roles: that of the
+client-facing server, as described in {{client-facing-server}}; or that of the
+backend server, as described in {{backend-server}}. The server plays the role of
+the client-facing server if it has custody of a set of ECH configurations.
+Otherwise, if the server does not have a set of ECH configurations, or if the
+`ClientECH` extension is empty, the server plays the role of the backend server.
```

Comment:
Good idea. Although I'm not sure what you mean here:
> {{backend-server}} probably wants to just use your second paragraph, and maybe clarify that the backend server just runs this logic unconditionally, so a potential client-facing server (the client may have just sent an empty ECH extension for no reason) and backend don't need to communicate much.

I updated the first paragraph of {{server-behavior}} with your suggestion and revised {{client-facing-server}} and {{backend-server}} so that the former applies only to non-empty values and the latter applies to empty values.

Reply:
Oh, that was a far too complicated suggestion on my part as to how to revise that text. Yours works too. :-)

The bit about "maybe clarify that [...]" was that we could optionally add something to the effect of:

> The backend server confirms acceptance based solely on the presence of an empty "encrypted_client_hello" extension. This allows the ECH negotiation to complete in Split Mode with minimal coordination between the client-facing server and backend server. Note, however, if the backend server is directly accessible to clients, it is possible for a client to send an empty "encrypted_client_hello" extension without a client-facing server decrypting a ClientHelloInner. The backend server will confirm acceptance in this case too.

(@dmcardle since we were talking about this odd implication of the ECH acceptance signal today.)